activesupport/lib/active_support/error_reporter.rb

```diff
@@ -108,13 +112,39 @@ def handle(*error_classes, severity: :warning, context: {}, fallback: nil, sourc
     #   source of the error. Subscribers can use this value to ignore certain
     #   errors. Defaults to <tt>"application"</tt>.
     def record(*error_classes, severity: :error, context: {}, source: DEFAULT_SOURCE)
-      error_classes = [StandardError] if error_classes.blank?
+      error_classes = DEFAULT_RESCUE if error_classes.empty?
       yield
     rescue *error_classes => error
       report(error, handled: false, severity: severity, context: context, source: source)
       raise
     end
 
+    # Either report the given error when in production, or raise it when in development or test.
+    #
+    # When raising, the original error is wrapped in a different error class to ensure
+    # it's not being rescued higher in the stack and will be surfaced to the developer.
+    #
```

Comment:
\+ explicit "when called in production, after the error is reported, this method will return nil and execution will continue", or similar?

"Code after the method will be executed if an exception is not raised" is clearly a little bit blatantly obvious... but I still feel like I want lots of giant flashing arrows around "every time you hit this code path in development it will do one thing, but in production something else will happen, so it's really important that whatever comes next isn't assuming some precondition has been enforced".

(e.g. code that looked like your example below, but without the `return`, would be Very Bad)